pkg/podexecutor/pebinary.go

```diff
@@ -0,0 +1,305 @@
+		"proxy-mode":        "kernelspace",
+		"kubeconfig":        p.cniConig.NodeConfig.AgentConfig.KubeConfigKubeProxy,
+		"network-name":      p.cniConig.NetworkName,
+		"bind-address":      p.cniConig.BindAddress,
+	}
+
+	if isDsrSupported() {
+		logrus.Infof("WinDSR support is enabled")
+		extraArgs["feature-gates"] = addFeatureGate(extraArgs["feature-gates"], "WinDSR=true")
+		extraArgs["enable-dsr"] = "true"
+	}
+
+	for range time.Tick(time.Second * 5) {
+		endpoint, err := hcsshim.GetHNSEndpointByName("Calico_ep")
+		if err != nil {
+			logrus.WithError(err).Warning("can't find %s, retrying", "Calico_ep")
+			continue
+		}
+		extraArgs["source-vip"] = endpoint.IPAddress.String()
+		break
+	}
+
+	logrus.Infof("Deleting HNS policies before kube proxy starts.")
+	policies, _ := hcsshim.HNSListPolicyListRequest()
+	for _, policy := range policies {
+		policy.Delete()
+	}
+
+	args = append(getArgsList(extraArgs), args...)
+	logrus.Infof("Running RKE2 Kubeproxy %s", args)
+	go func() {
+		for {
+			cmd := exec.Command(filepath.Join(p.DataDir, "bin", "kube-proxy.exe"), args...)
+			cmd.Stdout = os.Stdout
+			cmd.Stderr = os.Stderr
+			addDeathSig(cmd)
+			err := cmd.Run()
+			logrus.Errorf("Kubeproxy exited: %v", err)
+			time.Sleep(5 * time.Second)
+		}
+	}()
+
+	return nil
+}
+
+// APIServer sets up the apiserver static pod once etcd is available, returning the authenticator and request handler.
+func (p *PEBinary) APIServer(ctx context.Context, etcdReady <-chan struct{}, args []string) (authenticator.Request, http.Handler, error) {
+	panic("api server is unsupported on windows")
+}
+
+// Scheduler starts the kube-scheduler static pod, once the apiserver is available.
+func (p *PEBinary) Scheduler(apiReady <-chan struct{}, args []string) error {
+	panic("scheduler is unsupported on windows")
+}
+
+// ControllerManager starts the kube-controller-manager static pod, once the apiserver is available.
+func (p *PEBinary) ControllerManager(apiReady <-chan struct{}, args []string) error {
+	panic("controller manager is unsupported on windows")
+}
+
+// CurrentETCDOptions retrieves the etcd configuration from the static pod definition at etcd.yaml
+// in the manifests directory, if it exists.
+func (p *PEBinary) CurrentETCDOptions() (opts executor.InitialOptions, err error) {
+	panic("etcd options are unsupported on windows")
+}
+
+// CloudControllerManager starts the cloud-controller-manager static pod, once the cloud controller manager RBAC
+// (and subsequently, the api server) is available.
+func (p *PEBinary) CloudControllerManager(ccmRBACReady <-chan struct{}, args []string) error {
+	panic("cloud controller manager is unsupported on windows.")
+}
+
+// ETCD starts the etcd static pod.
+func (p *PEBinary) ETCD(args executor.ETCDConfig) error {
+	panic("etcd is unsupported on windows")
+}
+
+// isDsrSupport checks to determine if DSR can be enabled on windows or not. This is a required feature for Calico.
+func isDsrSupported() bool {
```

Comment:
I'm not sure you need your own implementation of this, upstream uses:
```golang
import "github.com/Microsoft/hcsshim/hcn"
hcn.DSRSupported()
```

Ref: https://github.com/kubernetes/kubernetes/blob/2423813207b885b59c19fddaa5eeb27cdfb8d44d/pkg/proxy/winkernel/proxier.go#L395